test/TestUtilities/Test.Utility/TestServer/PortReserverOfMockServer.cs

```diff
@@ -40,25 +41,35 @@ public class PortReserverOfMockServer : IDisposable
         /// </summary>
         /// <param name="basePath">The base path for all request URL's.
         /// Can be either null (default) for "/" or any "/"-prefixed string (e.g.:  /{GUID}).</param>
-        /// <param name="basePort">The base port for all request URL's.</param>
-        public PortReserverOfMockServer(string basePath = null, int basePort = 50231)
+        /// <param name="basePort">The base port for all request URL's. Defaults to system chosen available port,
+        /// at or below `65535`.</param>
+        public PortReserverOfMockServer(string basePath = null, int? basePort = null)
         {
             if (!string.IsNullOrEmpty(basePath) && (!basePath.StartsWith("/") || basePath.EndsWith("/")))
             {
                 throw new ArgumentException($"If provided, argument \"{nameof(basePath)}\" must start with and must not end with a slash (/)");
             }
 
-            if (basePort <= 0)
+            if (basePort is null)
             {
-                throw new InvalidOperationException();
+                // Port 0 means find an available port on the system.
+                var tcpListener = new TcpListener(IPAddress.Loopback, port: 0);
+                tcpListener.Start();
+                basePort = ((IPEndPoint)tcpListener.LocalEndpoint).Port;
```

Comment:
Consider using a different variable to store assigned port. I am not an expert in secure code, but, I think you can guess the value from the caller perspective.

Reply:
If there are security concerns, we should address those in a separate PR. We've been storing the port in a variable prior to this PR.

Reply:
Ok, it seems variable assignment just modifies the reference in call stack

Reply:
@NuGet/nuget-client could I get more input on whether this is exposing an available port on the call stack?